mypy/messages.py

```diff
@@ -332,6 +332,7 @@ def has_no_attr(self,
                 # The checker passes "object" in lieu of "None" for attribute
                 # checks, so we manually convert it back.
                 typ_format, orig_type_format = format_type_distinctly(typ, original_type)
+                # Not fully sure but I think we can replace "type(item) == NoneType" with "item is None"
```

Comment:
We can't... This is https://github.com/python/mypy/blob/fffbe88fc54807c8b10ac40456522ad2faf8d350/mypy/types.py#L698 , not `type(None)` (we could maybe use isinstance instead of `type(x) == ...`, but I'd prefer to just leave this alone)

Reply:
I think PEP recommends using `isinstance` over `type() == `. Would that be a better option?